activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -192,9 +192,14 @@ def compact_blank
   #   # => [ Person.find(1), Person.find(5), Person.find(3) ]
   #
   # If the +series+ include keys that have no corresponding element in the Enumerable, these are ignored.
-  # If the Enumerable has additional elements that aren't named in the +series+, these are not included in the result.
-  def in_order_of(key, series)
-    group_by(&key).values_at(*series).flatten(1).compact
+  # If the Enumerable has additional elements that aren't named in the +series+, these are not included in the result, unless
+  # the +filter+ option is set to +false+.
+  def in_order_of(key, series, filter: true)
+    if filter
+      group_by(&key).values_at(*series).flatten(1).compact
+    else
+      group_by(&key).values.flatten(1).sort_by { |v| series.index(v.public_send(key)) || series.size }.compact
```

Comment:
I think the following part is enough:
```suggestion
      sort_by { |v| series.index(v.public_send(key)) || series.size }.compact
```

Reply:
@willianveiga You're right. Ty!